GooglePlacesAutocomplete.d.ts

```diff
@@ -365,12 +370,29 @@ interface GooglePlacesAutocompleteProps extends TextInputProps {
   renderRightButton?: React.ComponentType<{}>;
 
   // sets the request URL to something other than the google api.  Helpful if you want web support or to use your own api.
-  requestUrl?: {
-    url: string;
-    useOnPlatform: 'web' | 'all';
-  };
+  requestUrl?: RequestUrl;
+}
+
+interface InitialState {
```

Comment:
this is an internal prop.  Does it need to be in the definitions file?  

Full disclosure: I am not all that familiar with typescript.

Reply:
I believe it is necessary to have access when using an own api as described in the comment above

Reply:
The library is not using typescript internally (it uses prop-types). Can you explain to me why it would still be necessary?

Reply:
Okay, but as I used to define the type of the parameter in the getRequestUrl method, which as I understand it is also public (without the underline prefix in front of the method name) I just added an interface to replace this raw object and use in both places.

Reply:
Let's try and keep this PR focused on solving the issues you encountered in #554.   

The public methods are notated as such in the code.  I am about to take off for the weekend and would love to get this merged and released.

Reply:
It's ok! Do you want me to remove all other changes and keep only solution # 554?

Reply:
Please keep the new interface for RequestUrl.  

Can you verify that adding `predefinedPlacesAlwaysVisible: boolean` clears the errors you were seeing?  I couldn't reproduce it myself.

Reply:
Yes, it clears the errors, but, reviewing the code now, I identified that in this line I did not put a question mark before the colon to define that this is optional, I will add it